### PR TITLE
Account for PredefinedCulturesOnly in CultureInfo LCID .ctor

### LINKED ISSUE #86878: PredefinedCulturesOnly=false not respected in new CultureInfo(int)
### Description

This creates issue for WinForms
https://github.com/dotnet/winforms/issues/9191#issuecomment-1567383463

see
https://github.com/dotnet/runtime/blob/182591af3ea67a09bca7e0516b950dafce71d5a6/src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.cs#L860-L872
and
https://github.com/dotnet/runtime/blob/182591af3ea67a09bca7e0516b950dafce71d5a6/src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.cs#L670-L682

### Reproduction Steps

`dotnet build /p:InvariantGlobalization=true /p:PredefinedCulturesOnly=false && dotnet run --no-build`
```c#
_ = new CultureInfo(1100);
```

### Expected behavior

Return Invariant culture.

### Actual behavior

Exception.

### PR DESCRIPTION
**Pull request by OwnageIsMagic**
Fixes #86878

### CODE REVIEW

**Comment by OwnageIsMagic** [comment]
@dotnet-policy-service agree

File: src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.cs [hunk 1]
```diff
@@ -868,6 +868,11 @@ internal static CultureData GetCultureData(int culture, bool bUseUserOverride)
 
             if (GlobalizationMode.Invariant)
             {
+                if (!GlobalizationMode.PredefinedCulturesOnly)
```
Comment (tarekgh) [inline src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.cs:871]: Please try adding some validation for the LCID and reject negative or 0 numbers.

**Comment by tarekgh** [comment]
@OwnageIsMagic could you please add a test case in the file https://github.com/dotnet/runtime/blob/main/src/libraries/System.Globalization/tests/Invariant/InvariantMode.cs?

File: src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.cs (hunk 1 above)
Comment (OwnageIsMagic) [reply src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.cs:871]: Should I move it to the main path?

**Comment by tarekgh** [comment]
@OwnageIsMagic I left a small suggestion, LGTM otherwise.

File: src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.cs [hunk 2]
```diff
@@ -868,6 +868,16 @@ internal static CultureData GetCultureData(int culture, bool bUseUserOverride)
...
+                {
+                    if (culture is < 1 or > 0xf_ffff)
+                    {
+                        throw new CultureNotFoundException(nameof(culture), culture, SR.Argument_CultureNotSupported);
+                    }
+
+                    return Invariant;
+                }
+
```
[the remark this answers is not in the document]
Comment (OwnageIsMagic) [reply src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.cs:880]: This doesn't seem like hot path, so I preferred readability. Also `Argument_CultureNotSupportedInInvariantMode` is misleading if range check fails and `Argument_CultureNotSupported` is more appropriate.

**Review by tarekgh — APPROVED** [review]

**Comment by OwnageIsMagic** [comment]
@tarekgh Thanks!

**Pull request merged**